src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -1069,6 +1092,36 @@ private bool CheckPropertyValueKind(SyntaxNode node, BoundExpression expr, BindV
             }
 
             return true;
+
+            bool isAllowedInitOnlySet(BoundExpression receiver)
+            {
+                // ok: new C() { InitOnlyProperty = ... }
+                if (receiver is BoundObjectOrCollectionValuePlaceholder)
+                {
+                    return true;
+                }
+
+                // bad: other.InitOnlyProperty = ...
+                if (!(receiver is BoundThisReference || receiver is BoundBaseReference))
+                {
+                    return false;
+                }
+
+                var containingMember = ContainingMemberOrLambda;
+                if (!(containingMember is MethodSymbol method))
+                {
+                    return false;
+                }
+
+                bool isInitOnlySetter = method.MethodKind == MethodKind.PropertySet && method.IsInitOnly;
```

Comment:
>method.MethodKind == MethodKind.PropertySet [](start = 40, length = 43)

Is this condition redundant?

Reply:
Yes, but it follows the letter of the spec, "Inside the init accessor of any property, on this or base"

---
In reply to: [415102012](https://github.com/dotnet/roslyn/pull/43275#discussion_r415102012) [](ancestors = 415102012)

Reply:
> Yes, but it follows the letter of the spec, "Inside the init accessor of any property, on this or base"

``` method.IsInitOnly``` already follows the letter of the spec

---
In reply to: [415204497](https://github.com/dotnet/roslyn/pull/43275#discussion_r415204497) [](ancestors = 415204497,415102012)

Reply:
Rename it to ```IsInitOnlySet``` if it is not obvious

---
In reply to: [415213918](https://github.com/dotnet/roslyn/pull/43275#discussion_r415213918) [](ancestors = 415213918,415204497,415102012)

Reply:
I don't think that's necessary.

---
In reply to: [415214044](https://github.com/dotnet/roslyn/pull/43275#discussion_r415214044) [](ancestors = 415214044,415213918,415204497,415102012)